webapp/app/[locale]/tunnel/_hooks/useEstimateDepositFees.ts

```diff
@@ -0,0 +1,40 @@
+import { encodeDepositErc20, encodeDepositEth } from 'hemi-tunnel-actions'
+import { useEstimateFees } from 'hooks/useEstimateFees'
+import { useL1StandardBridgeAddress } from 'hooks/useL1StandardBridgeAddress'
+import { EvmToken } from 'types/token'
+import { isNativeToken } from 'utils/nativeToken'
+import { useEstimateGas } from 'wagmi'
+
+export const useEstimateDepositFees = function ({
+  amount,
+  enabled = true,
+  fromToken,
+  toToken,
+}: {
+  amount: bigint
+  enabled?: boolean
+  fromToken: EvmToken
+  toToken: EvmToken
+}) {
+  const l1StandardBridge = useL1StandardBridgeAddress(fromToken.chainId)
+  const { data: gasUnits, isSuccess } = useEstimateGas({
```

Comment:
When calling `useEstimateGas` for an ETH deposit, `value` shall be sent too. For ERC-20 deposits, it is 0.

Reply:
Added here ! [2da1bd3](https://github.com/hemilabs/ui-monorepo/pull/1125/commits/2da1bd36c4bbf84fd8d32953cd6fc1c8c1dab13c)